lib/Resource.js

```diff
@@ -73,7 +73,7 @@ class Resource {
 		this._createStream = createStream || null;
 		this._stream = stream || null;
 		this._buffer = buffer || null;
-		if (string) {
+		if (typeof string === "string") {
```

Comment:
Should we also check for `string instanceof String` to catch cases where a `new String("bla")` object is passed?

Reply:
I would either use then both in combination:
typeof and instanceof

```js
let simpleStr = 'This is a simple string'
let myString  = new String()
let newStr    = new String('String created with constructor')

simpleStr instanceof String  // returns false, string literal is not an object
myString  instanceof String  // returns true
newStr    instanceof String  // returns true
(newStr + "a") instanceof String  // returns false
newStr.toString() instanceof String  // returns false
```
source + modified:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/instanceof

or leave typeof because that is rare that a string is constructed using the constructor.

Reply:
Yes, I would add that as an additional check as well. Just like here: https://stackoverflow.com/a/9436948

> or leave typeof because that is rare that a string is constructed using the constructor with new.
Yes it's rare, but if somebody does it we would still create the resource but without any content. That might cause issues that are not easy to understand. We should add this check to prevent this from happening.

Reply:
kk, done